### PR TITLE
feat: Integrated Eslint and Prettier

### LINKED ISSUE #75: Add Code Linter and Formatter to enhance code quality and maintain consistency.
### Is your feature request related to a problem? Please describe.
It is necessary for a project to have clear consistent coding conventions. Besides that linting is also very helpful for finding certain classes of bugs and fixing them before runtime.

### Describe the solution you'd like
[Eslint](https://eslint.org) and [Prettier](https://prettier.io) are by far the most popular code linter and formatter for Javascript.

* Integrate Eslint into the project
* Integrate Prettier into the project
* Make Eslint work with prettier i.e. Eslint for linting and Prettier for formatting code
* Add a brief description about them in the project readme. 
* Add Github workflows to ensure the guidelines are followed by every contributor.

### Describe alternatives you've considered
- [Rome](https://rome.tools/)
- We can also have a `.editorconfig` file in the project. EditorConfig's role is to configure the code editor so that the code we write is already formatted while Prettier will format the already written code.

### Additional context
WesBos maintains [this nice repo](https://github.com/wesbos/eslint-config-wesbos) related to eslint and prettier and we can take some ideas from there.

### PR DESCRIPTION
**Pull request by jalajcodes**
### Description
* Added Eslint into the project
* Added Prettier into the project
* Made Eslint work with prettier i.e. Eslint for linting and Prettier for formatting code
* Added a brief description about them in the project readme.
* Added Github workflows to ensure the guidelines are followed by every contributor

Fixes #75 

### Type of Change:

- Code
- Quality Assurance
- Documentation

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,6 +33,20 @@ npm start
...
+
+```sh
+npm run lint:fix
+```
+
+### To format files
+
+```sh
+npm run format
+```
```
Comment (ashwinijha6) [inline README.md]: Hey @jalajcodes you could remove the sh from ```sh that's not required

**Review by ashwinijha6 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jalajcodes) [reply README.md]: sure!

**Review by ashwinijha6 — CHANGES_REQUESTED** [review]
In docs/README.md which is that of docusaurus in the Deployment to surge part the line which mentions about surge there are certain mistakes **Tt is used to deploy our spen source** should be **It is used to deploy our open source** could you pls correct that.

**Comment by jalajcodes** [comment]
> In docs/README.md which is that of docusaurus in the Deployment to surge part the line which mentions about surge there are certain mistakes **Tt is used to deploy our spen source** should be **It is used to deploy our open source** could you pls correct that.

Okay so you weren't able to create the review for these lines coz I didn't change them in this PR, I think we can only create a review for the lines changed in the PR

anyways I'll correct it.

**Review by jhalak27 — COMMENTED** [review]
Look's good

**Comment by codesankalp** [comment]
@jalajcodes Why there are changes in markdown files? 
Can markdown files be skipped in this?

**Comment by jalajcodes** [comment]
> @jalajcodes Why there are changes in markdown files? 
> Can markdown files be skipped in this?

Yes, they can be ignored. Any particular reason you want to ignore them?

**Comment by codesankalp** [comment]
> Yes, they can be ignored. Any particular reason you want to ignore them?

As code linting is required so I thought markdown files can be skipped.

**Comment by jalajcodes** [comment]
> > Yes, they can be ignored. Any particular reason you want to ignore them?
> 
> As code linting is required so I thought markdown files can be skipped.

Okay I see....so should I revert markdown files back to their original state?

**Comment by codesankalp** [comment]
Yes, you can @jalajcodes but wait for @isabelcosta suggestion on this.
@isabelcosta Needs your review here.

**Comment by isabelcosta** [comment]
Up to you all. I noticed though, that improvements on markdown were kind of good. But I know this lint is more related to code. Is it possible to skip markdown @codesankalp 

I think there is not harm in leaving it for markdow. But whatever you decide here is fine

**Comment by codesankalp** [comment]
Ok, let's leave it to the current state, no need to make changes @jalajcodes.
I will review it soon.

File: .github/workflows/qa-checks.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+      # ESLint and Prettier must be in `package.json`
+      - name: Install Node.js dependencies
+        run: npm install
+
+      - name: Linters and Formatters Check
+        uses: wearerequired/lint-action@v1
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          # auto_fix: true
```
Comment (codesankalp) [inline .github/workflows/qa-checks.yml]: if this is not required you can remove this line. @jalajcodes

File: src/components/Profile.js [hunk 3]
```diff
@@ -1,187 +1,210 @@
...
-                    : <span>You have not filled in your basic information till now.</span>
-                }
-                </>
-            }
-            </>            
-        )
+  render() {
+    const { userinfo } = this.props;
+    {
+      userinfo && userinfo.length === 1
```
Comment (codesankalp) [inline src/components/Profile.js:62]: @jalajcodes Please create an issue for this to remove this warning also.

**Review by codesankalp — COMMENTED** [review]
Tested locally.
image

Code Review: Done
LGTM @jalajcodes.

File: .github/workflows/qa-checks.yml (hunk 2 above)
Comment (jalajcodes) [reply .github/workflows/qa-checks.yml]: Done!

File: src/components/Profile.js (hunk 3 above)
Comment (jalajcodes) [reply src/components/Profile.js:62]: Okay, I'll do that.

**Review by codesankalp — DISMISSED** [review]
Approved after testing again @jalajcodes.

@isabelcosta Needs your final review here.

After this PR do we need this https://github.com/anitab-org/open-source-programs-web/pull/68 as there is only a single warning showing which can be handled in the new issue.

**Review by Amulya-coder — COMMENTED** [review]
@jalajcodes please fix the conflicts.

**Comment by jalajcodes** [comment]
> @jalajcodes please fix the conflicts.

Done!

**Review by Amulya-coder — APPROVED** [review]

**Review by codesankalp — APPROVED** [review]

**Comment by codesankalp** [comment]
@isabelcosta I think this can be merged now.
Can you please review it again?

**Review by isabelcosta — APPROVED** [review]
Great work @jalajcodes !
Hopefully, this will help next changes, not requiring too many style changes 🤞🏾
Thank you for your work here!

**Pull request merged**